src/Serilog.Sinks.MongoDB/Sinks/MongoDB/MongoDBJsonFormatter.cs

```diff
@@ -80,6 +89,38 @@ public class MongoDBJsonFormatter : JsonFormatter
                 base.WriteJsonProperty(name, value, ref precedingDelimiter, output);
         }
 
+        private IReadOnlyDictionary<ScalarValue, LogEventPropertyValue> FixDictionary(IReadOnlyDictionary<ScalarValue, LogEventPropertyValue> dict)
+        {
+            var fixedDict = new Dictionary<ScalarValue, LogEventPropertyValue>();
+            foreach (var rec in dict)
+            {
+                var firstKey = rec.Key.Value;
+                string szKey = firstKey as string;
+
+                if (szKey != null)
+                    firstKey = szKey.Replace('$', '_').Replace('.', '-');
+
+                if (rec.Value.GetType() == typeof(DictionaryValue))
+                {
+                    var newValue = FixDictionary(((DictionaryValue)rec.Value).Elements);
+                    fixedDict[new ScalarValue(firstKey)] = new DictionaryValue(newValue);
+                    continue;
+                }
+                else
+                {
+                    if (rec.Value.GetType() == typeof(ScalarValue) && ((ScalarValue)rec.Value).Value.GetType() == typeof(string))
```

Comment:
```csharp
if (rec.Value is ScalarValue sv && sv.Value is string str)
```

Reply:
Thank you, I didn't know you could do that. Updated commit.

Reply:
Oh. Well, it broke the CI build when I added that part, but I think that would have to do with the compiler, it is complaining about all of the type patterns.

Reply:
Usually needs something like `<LangVersion>7.3</LangVersion>` in the CSPROJ